Fix sentence splitter: sentences ending with acronyms

nickto opened the pull request:
A sentence ending with an acronym was dot distinguished from an initial
followed by a period in the middle of the sentence. E.g., "Adamson is not from USA. 
They are from Europe" was considered as a single sentence because "USA." was 
treated as an initial.

1. Added a requirement for the initial to be only one letter long: "A." in "A. 
Adamson" is still treated as an initial, while "USA" in "from USA. They" is not.
2. Added a test for such case.

nickto committed:
Fix sentence splitter: sentences ending with acronyms

Sentence ending with an acronym was dot distinguisged from an initial
followed by a period in the middle of the sentence. Added a requirement
for the initial to be only one letter long, and added a test for such a
case.

oxinabox commented:
Good catch, do you think it would be better use use `\b` rather than `\s` ?
`\b` is  word-boundry, where as `\s` is white-space.
I think `\b` would work better if at the start of the string,

nickto committed:
Fix the double initials sentence splitting

Incorporated the following suggestion and added tests:
https://github.com/JuliaText/WordTokenizers.jl/pull/24#issuecomment-491307904

"This is A. A. Adamson." should be a single sentence, but older regex
would split it into mutiple because it required a space before the
second A.

nickto commented:
> Good catch, do you think it would be better use use `\b` rather than `\s` ?
> `\b` is word-boundry, where as `\s` is white-space.
> I think `\b` would work better if at the start of the string,

Changed it to `\b` and added a test that fails when using `\s`, thanks!

oxinabox commented:
LGTM, will merge when tests pass

oxinabox commented:
thank!
